readme.md

```diff
@@ -1298,11 +1298,13 @@ When reading from response stream fails.
 
 #### got.ParseError
 
-When server response code is 2xx, and parsing body fails. Includes a `response` property.
+> *ok*: 2xx or 304, and includes other 3xx if `options.followRedirect` is set
+
+When server response code is ***ok***, and parsing body fails. Includes a `response` property.
```

Comment:
This actually does not affect `ParseError`.

Reply:
@szmarczak Not really. See source/as-promise/index.ts.

```
// Parse body
try {
    response.body = parseBody(response, options.responseType, options.encoding);
} catch (error) {
    // Fallback to `utf8`
    response.body = rawBody.toString();

    if (isOk()) {
        // TODO: Call `request._beforeError`, see https://github.com/nodejs/node/issues/32995
        reject(error);
        return;
    }
}
```

Reply:
It doesn't have any effect. This code is **not** executed on redirects. `304` are empty responses, so even with `responseType: 'json'` it never throws:

https://github.com/sindresorhus/got/blob/ece94edec31bc8d004f93d74e218d8fd3b442c15/source/as-promise/core.ts#L28-L30

There is no `response` event emitted on redirect:

https://github.com/sindresorhus/got/blob/ece94edec31bc8d004f93d74e218d8fd3b442c15/source/core/index.ts#L1032-L1114